code/client/src/components/DomainPurchaseModal.jsx

```diff
@@ -0,0 +1,222 @@
+  <Space direction='vertical' style={WarningTextStyle}>
+    {
+      !enoughBalance && !checkingAvailability ? <Warning>Not enough ONE balance</Warning> : <></>
+    }
+    {
+      !domainAvailable && !checkingAvailability ? <Warning>Domain is not available</Warning> : <></>
+    }
+    {
+      checkingAvailability && validatedDomain ? <Spin /> : <></>
+    }
+  </Space>
+)
+
+/**
+ * Renders a modal that enables users to purchase an available domain for their selected wallet using selected token.
+ */
+const DomainPurchaseModal = ({ isModalVisible, dismissModal, oneBalance, walletAddress }) => {
+  const dispatch = useDispatch()
+
+  const [domainName, setDomainName] = useState('')
+
+  const [purchaseOnePrice, setPurchaseOnePrice] = useState(0)
+
+  const [domainFiatPrice, setDomainFiatPrice] = useState(0)
+
+  const [available, setAvailable] = useState(false)
+
+  const [enoughBalance, setEnoughBalance] = useState(false)
+
+  const [domainAvailable, setDomainAvailable] = useState(false)
+
+  const [checkingAvailability, setCheckingAvailability] = useState(true)
+
+  const price = useSelector(state => state.wallet.price)
+
+  const validatedDomain = validDomain(domainName)
+
+  const purchaseDomain = useCallback(async () => {
+    // The validated domain will be sent as [selectedDomainName].crazy.one.
+    dispatch(walletActions.purchaseDomain({ domainName: validatedDomain, address: walletAddress }))
+    dismissModal()
+  }, [domainName, walletAddress])
+
+  useWaitExecution(
+    async () => {
+      setCheckingAvailability(true)
+
+      const domainOnePrice = await api.blockchain.domain.price({ name: domainName })
+
+      const domainAvailability = await api.blockchain.domain.available({ name: domainName })
+
+      const computedDomainOnePrice = util.computeBalance(domainOnePrice.toString(), price)
+
+      const hasEnoughBalance = BigInt(domainOnePrice.toString()) <= BigInt(oneBalance)
+
+      const domainAvailableAndValid = domainAvailability && validatedDomain
+
+      setPurchaseOnePrice({ formatted: computedDomainOnePrice.formatted, value: domainOnePrice.toString() })
+
+      setDomainFiatPrice(computedDomainOnePrice.fiatFormatted)
+
+      setEnoughBalance(hasEnoughBalance)
+
+      setDomainAvailable(domainAvailableAndValid)
+
+      setAvailable(domainAvailableAndValid && hasEnoughBalance)
+
+      setCheckingAvailability(false)
+    },
+    validatedDomain,
+    delayCheckMillis,
+    [domainName, validatedDomain]
+  )
+
+  useEffect(() => {
+    if (!validatedDomain) {
+      setEnoughBalance(false)
+
+      setDomainAvailable(false)
+
+      setAvailable(false)
+
+      setCheckingAvailability(true)
+
+      setPurchaseOnePrice({ formatted: '0', value: '0' })
+
+      setDomainFiatPrice('0')
+    }
+  }, [validatedDomain, setEnoughBalance, setDomainAvailable, setAvailable, setPurchaseOnePrice, setDomainFiatPrice])
+
+  const onDomainName = (e) => {
+    setDomainName(e.target.value)
+  }
+
+  return (
+    <Modal
```

Comment:
Let's use flat components (e.g. Card) that fits into the main layout (similar to how transfer, set recovery, etc. are displayed). Modal distracts the users too much and it would easily lead us to layer too many things and lose track of which is on top of which